tests/components/minio/test_minio.py

```diff
@@ -0,0 +1,202 @@
+        DOMAIN,
+        {
+            DOMAIN: {
+                CONF_HOST: 'localhost',
+                CONF_PORT: '9000',
+                CONF_ACCESS_KEY: 'abcdef',
+                CONF_SECRET_KEY: '0123456789',
+                CONF_SECURE: 'true',
+            }
+        }
+    )
+
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    assert 'Setup of domain minio took' in caplog.text
+
+    # Call services
+    await hass.services.async_call(
+        DOMAIN,
+        'put',
+        {
+            'file_path': '/tmp/some_file',
+            'key': 'some_key',
+            'bucket': 'some_bucket'
+        },
+        blocking=True
+    )
+    assert minio_client.fput_object.call_args == call(
+        'some_bucket', 'some_key', '/tmp/some_file'
+    )
+    minio_client.reset_mock()
+
+    await hass.services.async_call(
+        DOMAIN,
+        'get',
+        {
+            'file_path': '/tmp/some_file',
+            'key': 'some_key',
+            'bucket': 'some_bucket'
+        },
+        blocking=True
+    )
+    assert minio_client.fget_object.call_args == call(
+        'some_bucket', 'some_key', '/tmp/some_file'
+    )
+    minio_client.reset_mock()
+
+    await hass.services.async_call(
+        DOMAIN,
+        'remove',
+        {
+            'key': 'some_key',
+            'bucket': 'some_bucket'
+        },
+        blocking=True
+    )
+    assert minio_client.remove_object.call_args == call(
+        'some_bucket', 'some_key'
+    )
+    minio_client.reset_mock()
+
+
+async def test_minio_listen(hass, caplog, minio_client, minio_event_response):
+    """Test minio listen on notifications."""
+    minio_client.presigned_get_object.return_value = 'http://url'
+
+    events = []
+
+    @callback
+    def event_callback(event):
+        """Handle event callbback."""
+        events.append(event)
+
+    hass.bus.async_listen('minio', event_callback)
+
+    await async_setup_component(
+        hass,
+        DOMAIN,
+        {
+            DOMAIN: {
+                CONF_HOST: 'localhost',
+                CONF_PORT: '9000',
+                CONF_ACCESS_KEY: 'abcdef',
+                CONF_SECRET_KEY: '0123456789',
+                CONF_SECURE: 'true',
+                CONF_LISTEN: [{
+                    CONF_LISTEN_BUCKET: 'test'
+                }]
+            }
+        }
+    )
+
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    assert 'Setup of domain minio took' in caplog.text
+
+    while not events:
+        await asyncio.sleep(0)
+
+    assert 1 == len(events)
+    event = events[0]
+
+    assert DOMAIN == event.event_type
+    assert 's3:ObjectCreated:Put' == event.data['event_name']
+    assert '5jJkTAo.jpg' == event.data['file_name']
+    assert 'test' == event.data['bucket']
+    assert '5jJkTAo.jpg' == event.data['key']
+    assert 'http://url' == event.data['presigned_url']
+    assert 0 == len(event.data['metadata'])
+
+
+async def test_queue_listener():
```

Comment:
Do we need to test this specifically? Isn't this already tested in the test above?

Reply:
It's not the same test. This is a unit test, testing a particular functionality.
The other one is more of an integration test. So in a way, it is also tested there, but it is in no way the same test.

Reply:
It's the other test that is important. We don't really care that the implementation should be in a specific way. I think we can remove these tests. They just decrease robustness.

Reply:
It actually helps you to pin point the issue faster. Please enlighten me how more (or more granular) tests decrease robustness ...

Reply:
We can't change the implementation without changing the test. That makes the test brittle.

Reply:
What? That the actual point of the test. You never want to have implementation in the tests, as that just adds another place, where bug can be introduced ...

Reply:
I can see both sides of the argument, but I think what is important in the long run is maintainability. If the potential downsides of needing to maintain the test outweigh the current benefit of having the test, then we can omit it

Reply:
Removed the tests ...